redshift/provider.go

```diff
@@ -64,15 +66,17 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	}
 
 	log.Println("[INFO] Initializing Redshift client")
-	client := config.Client()
-	db := client.db
+	client, err := config.Client()
+	if err != nil {
+		return nil, err
+	}
 
-	//Test the connection
-	err := db.Ping()
+	db := client.db
 
-	if err != nil {
-		log.Println("[ERROR] Could not establish Redshift connection with credentials")
-		panic(err.Error()) // proper error handling instead of panic
+	// DB connection is not opened until the first query.
+	// Test the connection
+	if err = db.Ping(); err != nil {
+		return nil, fmt.Errorf("Redshift connection error: %v", err)
```

Comment:
I see the postgres provider uses this: https://github.com/hashicorp/errwrap
Could be worth looking at?

Reply:
I dont really know to be honest, I trust your judgement on it